src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -991,6 +991,29 @@
         (autosign-certificate-request! subject csr settings)
         (fs/delete (path-to-cert-request csrdir subject))))))
 
+(schema/defn ^:always-validate process-csr-deletion!
```

Comment:
After writing most of this guy I found [`delete-certificate!`](https://github.com/puppetlabs/puppet-server/pull/998/commits/a635f9fd21a04047f25433e21daad4a58601f361#diff-feeac80211305477061ba2a798669e8dR1208). It does a _very_ similar thing, deleting the cert request if it exists, it also deletes a signed cert if it exists, but doesn't return any information about what it did or didn't end up deleting (or why).

It seems like these should share code, but I also was surprised that these two endpoints actions weren't mutually exclusive... Folks' feedback would guide which method I'd collapsed into which -- eg.
1. make this simpler by reusing the existing `delete-certificate!` fn and not returning as much info,
2. remove the csr deletion from `delete-certificate!` and have the certificate_statuses endpoint call both `delete-certificate!` and this fn or
3. make the delete-certificate fn more like this one (though still remove the csr deletion) and return more info from the endpoint (we don't currently return the documented information -- though the [documented response](https://docs.puppetlabs.com/puppet/4.2/reference/http_api/http_certificate_status.html#delete) is a very Ruby specific).



I actually looked at the certificate_status delete endpoint code but didn't go to the library implementation assuming deleting a cert and deleting a cert request would be fundamentally different things if they had different API endpoints. It wasn't until a bad grep for my code that I ran across it and after seeing the implementation went back to the docs for that endpoint...